src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
This is a great idea EXCEPT for `It would be preferable to use an entirely separate method, `.

No its this method that should do it.